### PR TITLE
fix: oracle expansion — stat-change targets, move metadata, Max Move power tables (Phases 1-3)

### LINKED ISSUE #1328: bug(gen3-4): generic stat-change translator routes many foe-targeting status moves onto the user
Severity: high

Parent audit: #1186

Location:
- Gen 3 and Gen 4 still use the generic stat-change translator in:
  - [Gen3MoveEffects.ts](/home/uehlbran/projects/pokemon-lib-ts/packages/gen3/src/Gen3MoveEffects.ts#L167)
  - [Gen4MoveEffects.ts](/home/uehlbran/projects/pokemon-lib-ts/packages/gen4/src/Gen4MoveEffects.ts#L379)
- That translator maps `effect.target === "self"` to the attacker and anything else to the defender.
- However, canonical move data for many foe-targeting status drops encodes `effect.target: "self"` even when the move target is an opposing Pokemon. Examples from local move data across the generation bundles:
  - `growl`
  - `tail-whip`
  - `leer`
  - `metal-sound`
  - `screech`
- Example data pattern:
  - move target: `adjacent-foe` or `all-adjacent-foes`
  - effect: `{ type: "stat-change", ..., target: "self" }`

Expected:
- These moves should lower the opponent's stat stages, not the user's.

Actual:
- The generic translator interprets `target: "self"` literally and routes the stat drop to the attacker, which is incompatible with the canonical move data shape for these foe-targeting status moves.

Authority / research:
- Local move data in `packages/gen3/data/moves.json` and `packages/gen4/data/moves.json`
- Local generic translator code above
- Cartridge/Showdown behavior for the listed moves

How found:
- `#1186` audit of the generic move-effect translator after tracing the later-gen fallback gap.
- Cross-checking canonical move data against the translator logic showed the target-field convention for several foe-targeting status drops does not match the current mapping.

Notes:
- This is distinct from #1326. #1326 covers Gen 5-9 falling through to an empty base move-effect implementation. This issue covers the target-mapping bug in the still-active Gen 3/4 generic translator.

### LINKED ISSUE #1441: bug(data-importer/gen5-9): foe-targeting stat-drop status moves are imported as self-targeting
Severity: high

Parent audit: #1186

Location:
- Checked-in `packages/gen5-9/data/moves.json` encodes many foe-targeting status debuffs as `effect: { type: "stat-change", target: "self", ... }`.
- Concrete examples from the shipped data:
  - `growl`
  - `leer`
  - `tail-whip`
  - `fake-tears`
  - `feather-dance`
  - `metal-sound`
  - `noble-roar`
  - `tearful-look`
- A wider scan across Gen 5-9 also catches the same pattern on `baby-doll-eyes`, `captivate`, `charm`, `confide`, `cotton-spore`, `eerie-impulse`, `play-nice`, `scary-face`, `screech`, `smokescreen`, `string-shot`, `sweet-scent`, `tickle`, `tar-shot`, and others.
- The importer path responsible is the generic status-move boost translation in `tools/data-importer/src/import-gen.ts:515-522`, which always maps `move.boosts && !move.basePower` to `target: self` without consulting the move's actual target.

Expected:
- These moves should be imported as foe-targeting stat drops so the data matches cartridge/Showdown semantics.

Actual:
- The checked-in data says the move lowers the user's own stats even when the move target and description clearly indicate the opposing Pokemon.
- This remains a data bug even separate from the current Gen 5-9 runtime fallback gaps.

Authority / research:
- Showdown `data/moves.ts` definitions for the listed moves
- Local checked-in move data in `packages/gen5-9/data/moves.json`
- Importer translation in `tools/data-importer/src/import-gen.ts:515-522`

How found:
- `#1186` audit pass comparing shipped move effects to move descriptions/targets.
- Repo scripts identified all moves where:
  - `effect.type === "stat-change"`
  - `effect.target === "self"`
  - the local description says "Lowers the target ..."

Notes:
- This is the Gen 5-9 counterpart to #1328, which covers the active Gen 3-4 translator bug. Here the bad targeting is already baked into the generated data itself.

### LINKED ISSUE #1588: bug(data/gen5-9): many foe-targeting stat-change moves are serialized with effect.target='self'
Severity: high

Parent audit: #1186

Location:
- Scanning `packages/gen5-9/data/moves.json` for `category: "status"`, `effect.type: "stat-change"`, and foe-targeting move targets shows a large family where the serialized `effect.target` is still `"self"`.
- Representative examples from the generated data:
  - Gen 5: `charm`, `cotton-spore`, `fake-tears`, `flash`, `growl`, `leer`, `memento`, `metal-sound`, `scary-face`, `screech`, `sweet-scent`, `tickle`
  - Gen 6-7: same family plus `baby-doll-eyes`, `confide`, `eerie-impulse`, `noble-roar`, `play-nice`, `tearful-look`
  - Gen 8-9: same family plus `decorate`, `howl`, `tar-shot`, `spicy-extract`
- Example data rows include:
  - `packages/gen9/data/moves.json` `charm`: `target: "adjacent-foe"`, `effect.target: "self"`
  - `packages/gen9/data/moves.json` `growl`: `target: "all-adjacent-foes"`, `effect.target: "self"`
  - `packages/gen9/data/moves.json` `tearful-look`: `target: "adjacent-foe"`, `effect.target: "self"`
- Repo-wide searches across later-gen runtime code do not find compensating per-move overrides for this family.

Expected:
- Foe-targeting stat-drop moves should serialize their stat changes against the defender / foe side.
- Ally/normal-target support moves like `Decorate` and `Spicy Extract` should preserve the correct target semantics rather than collapsing to attacker-side stat changes.

Actual:
- The later-gen canonical move data records a broad family of non-self status moves with `effect.target = "self"`.
- If the generic stat-change effect path trusts the serialized effect target, these moves will apply boosts/drops to the wrong side.

Authority / research:
- Local generated move data in `packages/gen5/data/moves.json` through `packages/gen9/data/moves.json`.
- Showdown / cartridge behavior for the listed moves.

How found:
- `#1186` audit pass over generated move-effect data.
- Scan used:
  - `node -e '... if (m.category==="status" && m.effect?.type==="stat-change" && m.target.includes("foe") && m.effect.target==="self") ...'`
- Follow-up searches across runtime code found no later-gen handlers for the affected IDs.

Notes:
- This is distinct from `#1342` / `#1344`, which focus on move target-class collapse. Here the problem is the serialized `effect.target` inside stat-change move data itself.
- It is also distinct from `#1328`, which covered an older Gen 3-4 translator bug; this later-gen issue appears in the generated canonical move JSON for Gen 5+.

### LINKED ISSUE #1590: bug(data/gen6-9): ally-target and ally-side stat-boost moves collapse to self/foe metadata
Severity: high

Parent audit: #1186

Location:
- Later generated move data shows multiple ally/support stat-boost moves with metadata that no longer matches their real targeting semantics.
- Representative examples from `packages/gen6-9/data/moves.json`:
  - `aromatic-mist`: `target = "adjacent-ally"`, but `effect.target = "self"`
  - `coaching`: `target = "adjacent-ally"`, but `effect.target = "self"`
  - `decorate`: `target = "adjacent-foe"`, but the description says it raises the target's Attack and Sp. Atk by 2
  - `howl` in Gen 8-9: `target = "adjacent-foe"`, while the description says it raises the Attack of the user and all allies
- Existing later-gen runtime searches do not show compensating per-move handlers for these IDs.

Expected:
- `Aromatic Mist` and `Coaching` should boost the adjacent ally, not the user.
- `Decorate` should retain its support-target semantics rather than becoming foe-only metadata plus self-target stat changes.
- Gen 8-9 `Howl` should preserve its self-and-allies boost semantics rather than collapsing to a foe-target move record.

Actual:
- The canonical later-gen move data collapses this family into combinations of `effect.target = "self"` and incorrect move target classes.
- Without dedicated corrective handlers, these moves will either boost the wrong Pokemon or be impossible to target/use correctly.

Authority / research:
- Local generated move data in `packages/gen6/data/moves.json` through `packages/gen9/data/moves.json`.
- Showdown / cartridge behavior for `Aromatic Mist`, `Coaching`, `Decorate`, and `Howl`.

How found:
- `#1186` audit pass over status stat-change move metadata.
- Scans used:
  - `node -e '... for (const id of ["aromatic-mist","coaching","decorate","howl"]) ...'`
  - comparison of serialized `target` / `effect.target` against move descriptions.

Notes:
- Related to `#1336`, `#1344`, `#1353`, and `#1588`, but narrower: this issue isolates ally-target / ally-side stat-boost moves whose encoded target/effect metadata cannot express the intended support behavior.

### PR DESCRIPTION
**Pull request by uehlbran**
## Summary

- **Phase 1**: Fix the binary truth collapse in stat-change target mapper — `STAT_CHANGE_TARGET_MAP` now uses an exhaustive `Record` where unknown Showdown targets throw instead of silently defaulting. Adds `ally` to `CORE_MOVE_EFFECT_TARGETS` and `StatChangeEffect.target`. Fixes 307 foe-targeting pure stat-change moves across Gen 2-9 (growl, leer, tail-whip, screech, etc.) that had `effect.target="self"`. Fixes 61 damaging moves with self-lowering drawbacks (close-combat, draco-meteor, leaf-storm, overheat, superpower) reverted to `"self"`. Fixes Gen 1 move metadata (counter/bide category + contact flag; guillotine/horn-drill contact flag). Adds `allies→self` to importer `mapTarget()` so Howl won't be re-imported with wrong target.
- **Phase 2**: Expand mechanics oracle suite — adds category, accuracy, PP, and 6-flag checks (contact/sound/punch/bite/bullet/powder) to `compare-mechanics.ts`. Adds `mechanics-checks.test.ts` (11 tests) and `stat-change-target-mapping.test.ts` (14 tests) with source-derived exhaustiveness checks via `@pkmn/data`.
- **Phase 3**: Implement deferred ground truth case types — `abilityCheck`, `moveRecoilCheck`, `maxMovePowerCheck` now evaluated. Corrects Gen 8 Max Move power table thresholds per ERRATA item 20: correct dual-table boundaries (Fighting/Poison `<45→70…≥150→100`; standard `<45→90…≥150→150`). Updates all gen8 tests to authoritative Showdown values (Showdown `sim/dex-moves.ts:517-549`).

## Test plan

- [x] `npm run test` — all 14 packages pass (1260 gen8 tests, 117 oracle-validation tests)
- [x] `npm run oracle:fast` — no new disagreements across all gens
- [x] `npm run typecheck` — clean
- [x] `npm run verify:local` — passes

Closes #1588
Closes #1590
Closes #1328
Closes #1441

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added ally-target support for move effects.

* **Bug Fixes**
  * Corrected stat-change targeting across Gens 2–9 and restored several self-targeting secondaries.
  * Fixed Gen 1 move categories and contact flags (Counter, Bide, Guillotine, Horn Drill).
  * Adjusted Gen 8 Max Move power mappings per ERRATA #20.
  * Corrected Howl targeting for singles and prevented criticals on fixed-damage moves.

* **Tests / Validation**
  * Expanded oracle-validation, edge-case suites, and targeted unit/integration tests.

* **Chores**
  * CI/workflow and proof/control-plane validation tooling updates.

### CODE REVIEW

File: tools/oracle-validation/src/proof-runner-output.ts [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+    skipped: 0,
+    incomplete: 0,
+    advisory: 0,
+  };
+}
+
+function inferRunMode(suitesRequested: readonly string[]): "fast" | "full" {
+  const requested = new Set(suitesRequested);
+  return requested.size === 1 && requested.has("fast") ? "fast" : "full";
+}
```
[the remark this answers is not in the document]
Comment (uehlbran) [reply tools/oracle-validation/src/proof-runner-output.ts]: Rechecked against the current branch: this one is already correct. `runner.ts` passes the original CLI `suites` array into `buildProofSummary(gitSha, suites, generationResults)`, while only `expandedSuites` is used for execution. That means a `--suite fast` request still reaches `inferRunMode` as `["fast"]`, so no fix was needed here.
